### PR TITLE
Adding Git information to test spans

### PR DESCRIPTION
**Pull request by tonyredondo**
This PR adds a git folder parser to include the git information to the test spans, as required in the ciapp specification.

Example: 
image

image

@DataDog/apm-dotnet

### CODE REVIEW

File: src/Datadog.Trace.Ci.Shared/CIEnvironmentValues.cs [hunk 1]
```diff
@@ -66,6 +81,21 @@ public static void DecorateSpan(Span span)
             span.SetTagIfNotNullOrEmpty(CommonTags.GitCommit, Commit);
             span.SetTagIfNotNullOrEmpty(CommonTags.GitBranch, Branch);
             span.SetTagIfNotNullOrEmpty(CommonTags.GitTag, Tag);
+            span.SetTagIfNotNullOrEmpty(CommonTags.GitCommitAuthorName, AuthorName);
+            span.SetTagIfNotNullOrEmpty(CommonTags.GitCommitAuthorEmail, AuthorEmail);
+            if (AuthorDate != DateTimeOffset.MinValue)
```
Comment (lucaspimentel) [inline src/Datadog.Trace.Ci.Shared/CIEnvironmentValues.cs:86]: Should we use a nullable `DateTimeOffset` for `AuthorDate` and `CommitterDate`?

**Review by lucaspimentel — COMMENTED** [review]
There's a ton of logic here to parse the git files that has no tests. Were you planning on adding them in a separate PR?

File: src/Datadog.Trace.Ci.Shared/CIEnvironmentValues.cs (hunk 1 above)
Comment (tonyredondo) [reply src/Datadog.Trace.Ci.Shared/CIEnvironmentValues.cs]: You're right, done in b69d87d

**Comment by tonyredondo** [comment]
> There's a ton of logic here to parse the git files that has no tests. Were you planning on adding them in a separate PR?

Yes I plan to collect a couple of `git like` folders to run the tests, and submit another PR for that.

**Review by zacharycmontoya — APPROVED** [review]
I glossed over the implementations of `TryGetPackageOffset` and `TryGetFromPackageOffset`. But everything besides that, including the git objects and the config parsing, look good. It sounds like you are planning to add tests in a separate PR so I think this PR is good to go

**Review by lucaspimentel — APPROVED** [review]
Cool feature. When can we use it? 😆

**Pull request merged**